src/components/Form.js

```diff
@@ -51,8 +53,13 @@ export default class Form extends Component {
   }
 
   validate(formData, schema) {
-    const validator = new Validator();
-    return validator.validate(formData, schema || this.props.schema).errors;
+    const {validate} = this.props;
+    const {errors} = jsonValidate(formData, schema || this.props.schema);
+    const errorSchema = toErrorSchema(errors);
+    if (typeof validate === "function") {
+      return userValidate(validate, formData, errorSchema);
+    }
```

Comment:
userValidate will return an `errors` property, recomputed against the internally augmented `errorSchema`. 

I know :(